src/rpc/rpc_trait.rs

```diff
@@ -7,24 +7,22 @@ use crate::{
     rpc::types::{relaxed, BlockHashOrTag, BlockNumberOrTag, Syncing},
     sequencer::{reply, request::Call},
 };
-use jsonrpsee::{proc_macros::rpc, types::error::Error};
+use jsonrpsee::types::error::Error;
 
 /// TODO
 /// Add proper output structs as per spec.
 /// Add proper error code handling as per spec.
-#[rpc(client, server, namespace = "starknet")]
+#[async_trait::async_trait]
 pub trait RpcApi {
```

Comment:
Do we still need a trait for this? Can we not just use the impl?

Reply:
I imagine a trait is useful when mocking for tests. But at the moment there is no such need and I cannot imagine when an rpc mock server would be needed :thinking: (we're not developing an rpc _client_). Besides `jsonrpsee` does not reexport `async_trait` anymore so I had to add the dependency explicitly :facepalm:.

Removing the trait will definitely make things more _KISS_, and if we ever need a trait we can always reintroduce it ;). Thanks for noticing! :pray: